### PR TITLE
Improve WorkflowExecutionStartEventAttributes documentation

### PR DESCRIPTION
**Pull request by Spikhalskiy**
**What changed?**
Clarified and added some comments to `WorkflowExecutionStartEventAttributes`.

### CODE REVIEW

File: temporal/api/history/v1/message.proto [hunk 1]
```diff
@@ -67,11 +69,13 @@ message WorkflowExecutionStartedEventAttributes {
...
     temporal.api.failure.v1.Failure continued_failure = 12;
     temporal.api.common.v1.Payloads last_completion_result = 13;
-    // This is the run id when the WorkflowExecutionStarted event was written
+    // This is the run id when the WorkflowExecutionStarted event was written.
+    // Reset changes the execution runId, but preserves this field.
     string original_execution_run_id = 14;
     // Identity of the client who requested this execution
     string identity = 15;
     // This is the very first runId along the chain of ContinueAsNew and Reset.
+    // Workflow Retry creates a new chain.
```
Comment (dnr) [inline temporal/api/history/v1/message.proto:78]: Retry (and cron) preserve this and don't create a new chain

Comment (Spikhalskiy) [reply temporal/api/history/v1/message.proto:78]: Thanks! I thought workflow retries create new chain. Is there anything you are aware of that breaks the chain now then?

File: temporal/api/history/v1/message.proto [hunk 2]
```diff
@@ -67,11 +69,13 @@ message WorkflowExecutionStartedEventAttributes {
     temporal.api.enums.v1.ContinueAsNewInitiator initiator = 11;
     temporal.api.failure.v1.Failure continued_failure = 12;
     temporal.api.common.v1.Payloads last_completion_result = 13;
-    // This is the run id when the WorkflowExecutionStarted event was written
+    // This is the run id when the WorkflowExecutionStarted event was written.
+    // Reset changes the execution runId, but preserves this field.
```
Comment (Sushisource) [inline temporal/api/history/v1/message.proto:73]:
```suggestion
    // A workflow reset changes the execution run_id, but preserves this field.
```
I figure snake case looks a little more consistent in proto land

**Review by Sushisource — APPROVED** [review]

File: temporal/api/history/v1/message.proto (hunk 1 above)
Comment (dnr) [reply temporal/api/history/v1/message.proto]: In my mental model, continuing a chain is the "exceptional case", limited to retry, [old] cron, continue-as-new, and reset, and "everything else" creates a new chain. But I guess there isn't that much else: start workflow (duplicate or not), signal-with-start, start child workflow. I can't think of any others.

Comment (Spikhalskiy) [reply temporal/api/history/v1/message.proto]: Thanks for confirming. 
So basically only one of the explicit start commands (all versions of it) can lead to the creation of the new chain. Other operations always continue the chain.

**Pull request merged**